src/core/core.helpers.js

```diff
@@ -53,6 +53,16 @@ module.exports = function(Chart) {
 		}
 		return base;
 	};
+    // Adjust element index when min is defined
+	helpers.adjustIndex = function(element, chartConfig) {
+		var min = chartConfig.options.scales.xAxes[0].ticks.min;
+		var labels = chartConfig.data.labels;
+
+		if (!element._adjustedIndex && min) {
```

Comment:
the condition should check `min !== undefined`. I may need to rethink this adjustment because this "adjustment" in the index is very specific to the Barchart.